lxd/auth/authorization_objects.go

```diff
@@ -0,0 +1,208 @@
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/gorilla/mux"
+)
+
+func ObjectUser(userName string) string {
+	return fmt.Sprintf("%s:%s", ObjectTypeUser, url.PathEscape(userName))
+}
+
+func ObjectServer() string {
+	return fmt.Sprintf("%s:lxd", ObjectTypeServer)
+}
+
+func ObjectCertificate(fingerprint string) string {
+	return fmt.Sprintf("%s:%s", ObjectTypeCertificate, url.PathEscape(fingerprint))
+}
+
+func ObjectStoragePool(storagePoolName string) string {
+	return fmt.Sprintf("%s:%s", ObjectTypeStoragePool, url.PathEscape(storagePoolName))
+}
+
+func ObjectProject(projectName string) string {
+	return fmt.Sprintf("%s:%s", ObjectTypeProject, url.PathEscape(projectName))
+}
+
+func ObjectImage(projectName string, imageFingerprint string) string {
+	return fmt.Sprintf("%s:%s/%s", ObjectTypeImage, url.PathEscape(projectName), url.PathEscape(imageFingerprint))
+}
+
+func ObjectImageAlias(projectName string, aliasName string) string {
+	return fmt.Sprintf("%s:%s/%s", ObjectTypeImageAlias, url.PathEscape(projectName), url.PathEscape(aliasName))
+}
+
+func ObjectInstance(projectName string, instanceName string) string {
+	return fmt.Sprintf("%s:%s/%s", ObjectTypeInstance, url.PathEscape(projectName), url.PathEscape(instanceName))
+}
+
+func ObjectNetwork(projectName string, networkName string) string {
+	return fmt.Sprintf("%s:%s/%s", ObjectTypeNetwork, url.PathEscape(projectName), url.PathEscape(networkName))
+}
+
+func ObjectNetworkACL(projectName string, networkACLName string) string {
+	return fmt.Sprintf("%s:%s/%s", ObjectTypeNetworkACL, url.PathEscape(projectName), url.PathEscape(networkACLName))
+}
+
+func ObjectNetworkZone(projectName string, networkZoneName string) string {
+	return fmt.Sprintf("%s:%s/%s", ObjectTypeNetworkZone, url.PathEscape(projectName), url.PathEscape(networkZoneName))
+}
+
+func ObjectProfile(projectName string, profileName string) string {
+	return fmt.Sprintf("%s:%s/%s", ObjectTypeProfile, url.PathEscape(projectName), url.PathEscape(profileName))
+}
+
+func ObjectStorageBucket(projectName string, poolName string, bucketName string) string {
+	return fmt.Sprintf("%s:%s/%s/%s", ObjectTypeStorageBucket, url.PathEscape(projectName), url.PathEscape(poolName), url.PathEscape(bucketName))
+}
+
+func ObjectStoragePoolVolume(projectName string, poolName string, volumeType string, volumeName string) string {
+	return fmt.Sprintf("%s:%s/%s/%s/%s", ObjectTypeStoragePoolVolume, url.PathEscape(projectName), url.PathEscape(poolName), url.PathEscape(volumeType), url.PathEscape(volumeName))
+}
+
+// ObjectByType returns an object of the given type. The passed in arguments must be in the correct
+// order (as found in the URL for the resource). This function will error if an invalid object type is
+// given, or if the correct number of arguments is not passed in. Project objects use the "projectName" argument.
+func ObjectByType(objectType ObjectType, projectName string, args ...string) (string, error) {
```

Comment:
Could you add an example here for a valid "args" slice in the docblock here?

Reply:
This has been removed.